modules/sitemaps/sitemaps-provider-decorator.php

```diff
@@ -59,29 +79,113 @@ public function get_url_list( $page_num, $object_subtype = '' ) {
 	/**
 	 * Gets the max number of pages available for the object type.
 	 *
-	 * Returns 0 for all languages but the default language for untranslated post types
-	 * and untranslated taxonomies to avoid to duplicate the corresponding sitemaps.
-	 *
 	 * @since 2.8
 	 *
 	 * @param string $object_subtype Optional. Object subtype. Default empty.
 	 * @return int Total number of pages.
 	 */
 	public function get_max_num_pages( $object_subtype = '' ) {
-		switch ( $this->provider->name ) {
-			case 'posts':
-				if ( ! PLL()->model->is_translated_post_type( $object_subtype ) && PLL()->model->options['default_lang'] !== $this->lang ) {
-					return 0;
-				}
-				break;
-			case 'taxonomies':
-				if ( ! PLL()->model->is_translated_taxonomy( $object_subtype ) && PLL()->model->options['default_lang'] !== $this->lang ) {
-					return 0;
-				}
-				break;
+		return $this->provider->get_max_num_pages( $object_subtype );
+	}
+
+	/**
+	 * Get active languages for the sitemap.
+	 *
+	 * @since 2.8
+	 */
+	protected function get_active_languages() {
+		$languages = $this->model->get_languages_list();
+		if ( wp_list_filter( $languages, array( 'active' => false ) ) ) {
+			return wp_list_pluck( wp_list_filter( $languages, array( 'active' => false ), 'NOT' ), 'slug' );
 		}
+		return wp_list_pluck( $languages, 'slug' );
+	}
```

Comment:
You can take advantage of the Decorator pattern here to move this into Polylang Pro. You just need to create a second decorator, that decorates this `get_active_languages()` method, and hook its instantiation **after** the instantiation of this one.

Here the function will only call `$this->model->get_languages_list()`, but Polylang Pro's decorator will call its decorated instance's (which is the Polylang Decorator) `get_active_languges()` method and then filter to only get active languages.

Reply:
I will leave it here for now to avoid adding complexity.